Gmcp tracking

awitteio opened the pull request:
Allows proper tracking of class resources from gmcp to svof.

- Created a new location to store class related info
- Altered back-end stuff to relate to the new location of class info
- Added in each classes proper tracking 

1. Alchemist (Wellspring)
2. Apostate (Essence)
3. Bard (Voice Balance, Stance)
4. Blademaster (Shin, Stance)
5. Depthswalker (Word Balance, Age)
6. Druid (Sunlight, Morph)
7. Infernal (Essence, Spec, Ferocity, Momentum)
8. Magi (Elementalism buffs)
9. Monk (Path, Stance, Kata, Kai)
10. Occultist (Karma, Entity)
11. Paladin (Devotion, Spec, Ferocity, Momentum)
12. Pariah (Epitaph Length)
13. Priest (Devotion, Conviction, Prayer Length, Prayer Balance)
14. Runewarden (Spec, Ferocity, Momentum)
15. Sentinel (Morph)
16. Serpent (Secreted Venom)
17. Shaman (Spirits)
18. Sylvan (Sunlight, Weatherweaving Buffs)
19. EartLord (Shaping Count)
20. Fire (Spark Count)

Commit:
Update svo (limbcounter).xml

Commit:
Update svo (actions dictionary).xml

Commit:
Merge branch 'in-client-svof' of https://github.com/K1ngSpad3z/svof into in-client-svof

Commit:
Changes to Backend Tracking

- These changes allow svof to track all class related resources through gmcp and updates them through signals. Things tracked are class balances (Voice balance for bards, word balance for depthswalkers, etc), class resources (devotion, essence, sunlight, etc), and other things (occultist entity, monk stance, etc) reliably without having to rely on triggers and such to do so

DazedxJoker commented on svo (setup, misc, empty, funnies, dor).xml:
```diff
@@ -143,6 +143,8 @@ local gaffl          = svo.gaffl
 svo.gdefc            = svo.gdefc or {}
 svo.me               = svo.me or {}
 local me             = svo.me
+svo.classinfo        = {}
```
Is there a reason this entry isn't svo.classinfo = svo.classinfo or {}? 

Would empty the table on saving the script.

awitteio committed:
Typo fix

Commit:
More Changes

Custom Prompt broke. Had to change some values to compare to itself properly as well as set the global variable

Commit:
Merge branch 'GMCP-Tracking' of https://github.com/K1ngSpad3z/svof into GMCP-Tracking

Commit:
More Fixes for Custom Prompt

forgot to change age to new location

tsartin commented on svo (limbcounter).xml:
```diff
@@ -2186,7 +2186,7 @@ if table.contains(svo.lc_classlist, svo.me.class) or svo.defc["dragonform"] == t
       })
       svo.echof("Loaded svo limbcounter, %s %s.", "Monk Tekura", tostring(svo.modules_version["svo (limbcounter)"]))
-    elseif svo.me.path == "Shikudo" then
+    elseif svo.classinfo.path == "Shikudo" then
```
I believe this "Shikudo" should be "shikudo"

tsartin commented on svo (custom prompt, serverside, peopletracker).xml:
```diff
@@ -317,7 +317,7 @@ cpp.compute_gametargethp = function()
+  return classinfo.weaponmastery or 0
 end
 
 cpp.compute_power = function()
```
Line 324 should be changed from 'stats.shin or stats.kai' to 'classinfo.shin or classinfo.kai'

tsartin commented on svo (limbcounter).xml:
```diff
@@ -2186,7 +2186,7 @@ if table.contains(svo.lc_classlist, svo.me.class) or svo.defc["dragonform"] == t
+    elseif svo.classinfo.path == "Shikudo" then
       enableTrigger("svo Monk limbcounter")
       enableTrigger("Shikudo")
       disablTrigger("Tekura")
```
typo here. 'disableTrigger'

tsartin commented on svo (setup, misc, empty, funnies, dor).xml:
```diff
@@ -386,128 +389,655 @@ do
+  signals.gmcpcharvitals:connect(function()
+    if gmcp.Char.Vitals.charstats then
+      for _, val in ipairs(gmcp.Char.Vitals.charstats) do
+        local essence = val:match("^Essence: (%d+)%%$")
```
I'm not 100% sure but is the regex correct here? Seems like an extra %.

tsartin commented on svo (setup, misc, empty, funnies, dor).xml:
```diff
@@ -386,128 +389,655 @@ do
+    if not svo.classinfo.spec then
+       svo.classinfo.spec = "None"
+    end
+  end, 'update knightspec')
```
I'm not sure if this is a typo or on purpose. 'update knightspec' vs 'update knight spec'.

All the knights have the the same so may not matter.

Sidenote: Achaea refers to these classes as 'warriors' now if that matters.

tsartin reviewed (changes requested):
I'd love to see this finished but I'm also not sure how this will affect the changes that have been made since this was created. For example, there have been new classes added to svo.lc_classlist. There is now stats.manifestation and stats.karma which would need to be changed.

TheLastDarkthorne commented:
@K1ngSpad3z I remember that changing the stats from the usual location was causing troubles before... would you mind to have the gmcp tracking of this stuff put the variables in the usual place? Or maybe let me grab these to put into an update while you work on changing it to another location later?
